addons/web_editor/controllers/main.py

```diff
@@ -625,3 +625,15 @@ def save_library_media(self, media):
             attachments.append(attachment._get_media_info())
 
         return attachments
+
+    @http.route("/web_editor/name_search_read", type="json", auth="user", website=True)
+    def name_search_read(self, model, name='', args=None, operator='ilike', limit=100, fields=None):
+        """
+        Performs a name_search followed by a search_read of the desired fields
+        """
+        if not request.env.user.has_group('website.group_website_publisher'):
+            raise AccessError(_("You do not have access"))
+        Model = request.env[model]
+        records = Model.name_search(name, args, operator=operator, limit=limit)
+        ids = [record[0] for record in records]
+        return Model.browse(ids)._read_format(fields)
```

Comment:
May I ask about the high level rationale for introducing a specific route for "name_search_read"? Is it just to save a RPC? What sort of extra fields do you need to read in practice, considering that the  m2o autocompletion works just fine in the backend with just the vanilla `name_search()`?

I'd rather not have a proliferation of alternative routes for calling the basic model methods, as it will create inconsistencies, including for security. For example the extra `has_group()` is weird considering the `auth=user` + normal application of `name_search()` permissions.

Also, if this need is considered generic enough, it should perhaps not be a custom route in `web_editor`, but instead a method on `BaseModel` in the framework itself.

Reply:
@JKE-be @qsm-odoo one of the few parts of the original PR that I did not touch so I'm not too sure on the original reason. Here's the comment on the original PR: https://github.com/odoo/odoo/pull/45275#discussion_r380050630

Reply:
@sdegueldre Let's not break our brains following the python path of those `name_search` + `_read_format` to see if this was actually needed, I don't remember :wink: Try to go with the extra RPC as I suggested last year and see how it goes (so `name_search` + `search_read`). No need of the extra `has_group` check since you do it from the editor and maybe less efficient both client side and server side but if it works without an extra route... :+1: 

Note: I push-forced this branch before r+ last week so please fetch it before adapting :wink: Also try to rebase, push-force, adapt the work, then re-push-force so I can easily see the new changes. Thanks :+1:

Reply:
@qsm-odoo I think I fetched your changes before making the changes related to the use of `.html` tough I'm not sure anymore, any chance you remember what you changed? Diffing the commits before and after your pushes wasn't very helpful